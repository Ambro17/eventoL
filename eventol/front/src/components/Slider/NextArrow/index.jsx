import React from 'react'
import PropTypes from 'prop-types'

const NextArrow = ({onClick, style}) => (
  <div
    className='arrow arrow-next fa fa-chevron-right'
    style={{...style, display: 'block', color: 'black'}}
    onClick={onClick} />
);

NextArrow.propTypes = {
  onClick: PropTypes.func,
  style: PropTypes.object
};

export default NextArrow;
