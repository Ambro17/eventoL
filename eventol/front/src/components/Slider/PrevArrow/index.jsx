import React from 'react'
import PropTypes from 'prop-types'

const PrevArrow = ({onClick, style}) => (
  <div
    className='arrow arrow-prev fa fa-chevron-left'
    style={{...style, display: 'block', color: 'black'}}
    onClick={onClick} />
);

PrevArrow.propTypes = {
  onClick: PropTypes.func,
  style: PropTypes.object
};

export default PrevArrow;
