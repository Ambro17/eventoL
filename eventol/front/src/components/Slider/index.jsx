import React from 'react'
import PropTypes from 'prop-types'
import ReactSlickSlider from 'react-slick'

import NextArrow from './NextArrow';
import PrevArrow from './PrevArrow';
import {SLIDER_BASE_SETTINGS} from '../../utils/constants';

import './index.css';


const Slider = ({children:items}) => {
  const settings = {
    ...SLIDER_BASE_SETTINGS,
    prevArrow: <PrevArrow />,
    nextArrow: <NextArrow />
  }
  return (
    <ReactSlickSlider className='container' {...settings}>
      {items}
    </ReactSlickSlider>
  );
};

Slider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
    PropTypes.element
  ])
};

export default Slider;
