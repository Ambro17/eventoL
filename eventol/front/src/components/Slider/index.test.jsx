import React from 'react';
import renderer from 'react-test-renderer';

import Slider from '.';

describe('Slider', () => {
  let component;
  let instance;
  let tree;

  const getComponent = () => {
    component = renderer.create(<Slider />);
    return component;
  };

  beforeEach(() => {
    component = getComponent();
    instance = renderer.root;
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
