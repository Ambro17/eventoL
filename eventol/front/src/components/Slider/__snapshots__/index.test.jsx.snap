// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Default Snapshot 1`] = `
<div
  className="slick-slider container slick-initialized"
>
  
  <div
    className="slick-list"
  >
    <div
      className="slick-track"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "WebkitTransform": "translate3d(0px, 0px, 0px)",
          "WebkitTransition": "",
          "msTransform": "translateX(0px)",
          "opacity": 1,
          "transform": "translate3d(0px, 0px, 0px)",
          "transition": "",
        }
      }
    />
  </div>
  
  
</div>
`;
